groovy-language-server/src/test/java/com/palantir/ls/groovy/GroovyLanguageServerIntegrationTest.java

```diff
@@ -0,0 +1,419 @@
+                                        "unable to resolve class ExceptionNew1 \n @ line 7, column 18.",
+                                        DiagnosticSeverity.Error)
+                                                .range(Ranges.createRange(6, 17, 6, 72))
+                                                .source(test1.getAbsolutePath())
+                                                .build())),
+                        new MyPublishDiagnosticParams(test2.toPath().toUri().toString(),
+                                Sets.newHashSet(new DefaultDiagnosticBuilder(
+                                        "unable to resolve class ExceptionNew222 \n @ line 7, column 18.",
+                                        DiagnosticSeverity.Error)
+                                                .range(Ranges.createRange(6, 17, 6, 74))
+                                                .source(test2.getAbsolutePath())
+                                                .build())));
+        assertEquals(expectedDiagnosticsResult, publishedDiagnostics.stream().collect(Collectors.toSet()));
+        assertEquals(2, publishedDiagnostics.size());
+    }
+
+    private void sendDidOpen(File file) {
+        client.getTextDocumentService()
+                .didOpen(new DidOpenTextDocumentParamsBuilder()
+                        .textDocument(
+                                new TextDocumentItemBuilder()
+                                        .languageId("groovy")
+                                        .uri(file.getAbsolutePath())
+                                        .version(0)
+                                        .text("foo")
+                                        .build())
+                        .build());
+    }
+
+    private void assertCorrectInitializeResult(InitializeResult result) {
+        ServerCapabilities expectedCapabilities = new ServerCapabilitiesBuilder()
+                .textDocumentSync(TextDocumentSyncKind.Incremental)
+                .documentSymbolProvider(true)
+                .workspaceSymbolProvider(true)
+                .referencesProvider(true)
+                .completionProvider(new CompletionOptionsBuilder()
+                        .resolveProvider(false)
+                        .triggerCharacter(".")
+                        .build())
+                .build();
+        assertEquals(Sets.newHashSet(
+                new LanguageDescriptionBuilder()
+                        .languageId(GroovyConstants.GROOVY_LANGUAGE_NAME)
+                        .fileExtension(GroovyConstants.GROOVY_LANGUAGE_EXTENSION)
+                        .build()),
+                Sets.newHashSet(result.getSupportedLanguages()));
+        ServerCapabilities capabilities = result.getCapabilities();
+        assertEquals(expectedCapabilities, capabilities);
+    }
+
+    private static File addFileToFolder(File parent, String filename, String contents) throws IOException {
```

Comment:
Because it doesn't need to be non-static. It doesn't use any of the instance variable or functions.